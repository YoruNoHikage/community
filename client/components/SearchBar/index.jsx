import React, { useState } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  results: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      render: PropTypes.func.isRequired,
    })
  ).isRequired,
  loading: PropTypes.bool.isRequired,
  onSelect: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired,
};

const SearchBar = ({ results, loading, onSelect, onSearch }) => {
  return (
    <div className="dropdown is-active" style={{ display: 'flex' }}>
      <div
        className={`control is-expanded ${loading ? 'is-loading' : ''}`}
        style={{ flex: 1 }}
      >
        <input
          className="input is-rounded"
          type="text"
          placeholder="Search"
          onChange={e => {
            onSearch(e.target.value);
          }}
        />
      </div>
      {results && results.length > 0 && (
        <div className="dropdown-menu" role="menu">
          <div className="dropdown-content">
            {results.map(item => (
              <a
                key={item.key}
                role="button"
                className="dropdown-item"
                onClick={() => {
                  onSelect(item.key);
                }}
              >
                {item.render()}
              </a>
            ))}
          </div>
        </div>
      )}
    </div>
  );
};

export default SearchBar;
