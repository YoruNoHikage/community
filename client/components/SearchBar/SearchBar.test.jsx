import React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from '.';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <SearchBar
        results={[]}
        loading
        onSearch={() => null}
        onSelect={console.log}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
