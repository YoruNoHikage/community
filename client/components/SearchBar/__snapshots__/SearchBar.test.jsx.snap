// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dropdown is-active"
>
  <div
    className="control is-expanded is-loading"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <input
      className="input is-primary"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </div>
</div>
`;
